sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -432,4 +435,18 @@ class DatasetAggregatorSuite extends QueryTest with SharedSparkSession {
     val agg = df.select(mode(col("a"))).as[String]
     checkDataset(agg, "3")
   }
+
+  test("Reduce aggregator scala input types") {
```

Comment:
I am not sure what we are testing here. The problem is that the current API does not allow this to happen. So unless we change the API this test is a bit too contrived for my taste.